src/Format.cpp

```diff
@@ -11,42 +11,35 @@
 
 ByteString format::URLEncode(ByteString source)
 {
-	char * src = (char *)source.c_str();
-	char * dst = new char[(source.length()*3)+2];
-	std::fill(dst, dst+(source.length()*3)+2, 0);
+	ByteString finalString;
 
-	char *d;
-	for (d = dst; *d; d++) ;
-
-	for (unsigned char *s = (unsigned char *)src; *s; s++)
+	for (auto& s : source)
```

Comment:
No good reason to iterate by reference here (left alone by mutable reference). `char s : source` would be better.